src/core/ext/filters/client_channel/retry_throttle.h

```diff
@@ -61,16 +63,22 @@ class ServerRetryThrottleData : public RefCounted<ServerRetryThrottleData> {
 /// Global map of server name to retry throttle data.
 class ServerRetryThrottleMap {
  public:
-  /// Initializes global map of failure data for each server name.
-  static void Init();
-  /// Shuts down global map of failure data for each server name.
-  static void Shutdown();
+  static ServerRetryThrottleMap* Get();
 
   /// Returns the failure data for \a server_name, creating a new entry if
   /// needed.
-  static RefCountedPtr<ServerRetryThrottleData> GetDataForServer(
+  RefCountedPtr<ServerRetryThrottleData> GetDataForServer(
       const std::string& server_name, intptr_t max_milli_tokens,
       intptr_t milli_token_ratio);
+
+ private:
+  using StringToDataMap =
+      std::map<std::string, RefCountedPtr<ServerRetryThrottleData>>;
+
+  static std::atomic<ServerRetryThrottleMap*> instance_;
```

Comment:
Can we just make this a local static pointer inside of the `Get()` method?  That way, I think it doesn't need to be atomic.

Reply:
It has a non-trivial destructor and so per style guide cannot be made static.

Reply:
Done